### PR TITLE
Fix the bug in interpreting the replicas of Job

### PR DESCRIPTION
**Pull request by whitewindmills**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-controller-manager`: Fixed the issue that the default resource interpreter doesn't accurately interpret the numbers of replicas.
```

### CODE REVIEW

File: pkg/resourceinterpreter/default/native/replica.go [hunk 1]
```diff
@@ -86,6 +87,13 @@ func jobReplica(object *unstructured.Unstructured) (int32, *workv1alpha2.Replica
 	if job.Spec.Parallelism != nil {
 		replica = *job.Spec.Parallelism
 	}
+	// For fixed completion count Jobs, the actual number of pods running in parallel will not exceed the number of remaining completions.
+	// Higher values of .spec.parallelism are effectively ignored.
+	// More info: https://kubernetes.io/docs/concepts/workloads/controllers/job/
+	completions := ptr.Deref[int32](job.Spec.Completions, replica)
```
Comment (chaunceyjiang) [inline pkg/resourceinterpreter/default/native/replica.go:93]: Indeed. In my local test, when parallelism > completions, only completions number of jobs will be generated.

Comment (chaunceyjiang) [reply pkg/resourceinterpreter/default/native/replica.go:93]: image

image

**Review by chaunceyjiang — COMMENTED** [review]
/lgtm

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot
/lgtm

**Comment by whitewindmills** [comment]
ping @RainbowMango

File: pkg/resourceinterpreter/default/native/replica.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/resourceinterpreter/default/native/replica.go:93]: So, we are trying to address the accuracy issue. Right?

Generally looks good to me, just a question need to confirm.

Comment (whitewindmills) [reply pkg/resourceinterpreter/default/native/replica.go:93]: yes

**Review by RainbowMango — APPROVED** [review]
/approve

I updated the release notes by the way.

**Pull request merged**